### ISSUE TITLE
Autocomplete Issue with foreign keyboards (Korean)

### ISSUE THREAD

**Issue by TosinAF** [slackhq/SlackTextViewController#398]
Using `/` when typing in Korean on iOS forces letters to separate and not combine as they should (and this works as intended in other apps on iOS)

User Comment 
> In iPhone client, using Korean slash command is not available. When i type the first key of the command like “ㅇ” of “어디”, because of the autocompletion, cursor goes to next so i cannot type “어”. It make me type like “ㅇㅓ”. In Korean, two or three keys can be a one letter. We combine “ㅇ” and “ㅓ” to make “어”. The only way to use “/어디” is typing “ㅇ” and find “어디” in autocompletion list and choose it. Thank you :D

<img width="1123" alt="screen shot 2016-02-18 at 9 46 50 am" src="https://cloud.githubusercontent.com/assets/2206858/13154838/c60bf406-d62f-11e5-8c34-119cfd767bec.png">

**Comment by dzenbot** [comment]
That reminds me of https://github.com/slackhq/SlackTextViewController/blob/30f3ecc9926b08443f4a61d47f30a23d0f8d0aac/Source/SLKTextViewController.m#L1167-L1171
It's probably a thing for most east asian languages. Will dig in soon.

**Comment by TosinAF** [comment]
Could probably do the same thing for now and maybe focus on when we start to localize.

**Comment by dzenbot** [comment]
Yup. Although, this isn't really a localisation issue. More like an race condition by all the little hacks we do to disable the auto-correction for auto-completion, in east asian languages.

**Comment by dzenbot** [comment]
Closing this for now.

**Comment by intoxicated** [comment]
@dzenbot which part of code should I look to fix this problem? any ideas?

**Comment by intoxicated** [comment]
@TosinAF @dzenbot Actually it wasn't problem with SLK in my case. So I looked into every single method in SLK when autocompletion worked. And this didn't happened when `didChangeAutoCompletionPrefix:word` didn't get called which leaded me to my own source code where I overrided this method. So when this method was called, I originally did filtering/sorting stuffs to display autocompletion in main thread, and I gave a shot to run this on background since I doubted maybe these work load might cause to textview not display text correctly. I simply used GCD to do sorting/filtering work and called `showAutoCompletionView:` in main thread, and it worked as I expected!
